### DISCUSSION TITLE
HTTP 404 error when signing

### DISCUSSION THREAD

**Discussion by oleksii-tymofieiev** [ebourg/jsign#218] [Q&A] [answered]
Hello,

I am getting this error while signing:
`Caused by: java.io.IOException: HTTP Error 404 - Not Found (https://cloudkms.googleapis.com/v1/projects/<PROJECT_NAME>/locations/us-west2/keyRings/Code-Signing-Certs/cryptoKeys/DigiCert_EV-Code-Signing2/cryptoKeys/cryptoKeyVersions/1:asymmetricSign)`

Nothing to the best of my knowledge has been changed on our side since March when the same signing command was working.
The keys are available and listable in Google and in my console CLI.

Could I ask what specific resource is not available for this request?
Hitting this endpoint with curl also returns 404 so probably something is with the URL.

Thank you,
Oleksii.

**Comment by ebourg** [comment]
There is something wrong with the URL, `cryptoKeys` appears twice. The path should look like this:

    projects/<PROJECT_NAME>/locations/us-west2/keyRings/Code-Signing-Certs/cryptoKeys/DigiCert_EV-Code-Signing2/cryptoKeyVersions/1

Check the `--alias` parameter, it should be `DigiCert_EV-Code-Signing2/cryptoKeyVersions/1`

**Comment by oleksii-tymofieiev** [comment]
Have double-checked, the `alias` looks like it should be.

```
jsign --storetype GOOGLECLOUD \
       --keystore projects/<PROJECT_NAME>/locations/us-west2/keyRings/Code-Signing-Certs/cryptoKeys/DigiCert_EV-Code-Signing2 \
       --storepass $(gcloud auth application-default print-access-token) \
       --alias DigiCert_EV-Code-Signing2/cryptoKeyVersions/1 \
       --certfile ~/Cert-Signing/full_chain.pem ~/Downloads/app.exe
```

**Reply by ebourg** [reply, marked as answer]
The alias is good, but the keystore isn't, it should not include the name of the key.

Try: `--keystore projects/<PROJECT_NAME>/locations/us-west2/keyRings/Code-Signing-Certs`

**Reply by ebourg** [reply]
The keystore parameter is now validated (https://github.com/ebourg/jsign/commit/d21079ba7d1f0f07638a1a485ddea5d82ed80e79), and in this case a useful error message is now displayed.

**Comment by oleksii-tymofieiev** [comment]
Thanks @ebourg this seems to be the answer.
